protocols/request-response/src/handler.rs

```diff
@@ -313,7 +313,7 @@ where
             self.outbound.shrink_to_fit();
         }
 
-        if self.inbound.is_empty() {
+        if self.inbound.is_empty() && self.keep_alive.is_yes() {
```

Comment:
Paraphrasing this to make sure I am on the same page: Without the additional `&& self.keep_alive.is_yes()` this condition would be triggered on each poll if the `inbound` queue is empty and thus the `self.keep_alive` timeout would be refreshed each time leaving it no time to expire. Is that correct?

Reply:
That is correct.